reconcile/terraform_cloudflare_dns.py

```diff
@@ -199,6 +228,24 @@ def are_record_identifiers_duplicated_within_zone(
     return duplicate_exist
 
 
+def ensure_record_number_not_exceed_max(
+    zones: list[CloudflareDnsZoneV1], default_max_records: int
+) -> None:
+    for zone in zones:
+        if not zone.records:
+            continue
+        num_records = len(zone.records)
+        if not zone.max_records:
+            zone.max_records = default_max_records
```

Comment:
Any reason not track this in a separate variable? If you mutate the `zone` data, could that cause confusion for someone debugging this later, maybe in other parts of the code where it isn't obvious it was changed here?